libusb/os/windows_winusb.c

```diff
@@ -1387,9 +1388,14 @@ static int set_composite_interface(struct libusb_context *ctx, struct libusb_dev
 	// devices will have only MI_00 & MI_03 for instance), we retrieve the actual
 	// interface number from the path's MI value
 	mi_str = strstr(device_id, "MI_");
-	if ((mi_str != NULL) && isdigit((unsigned char)mi_str[3]) && isdigit((unsigned char)mi_str[4])) {
-		interface_number = ((mi_str[3] - '0') * 10) + (mi_str[4] - '0');
-	} else {
+        // Initialize to default
+        interface_number = 0;
+		errno = 0;
+		char* endptr = NULL;
+	if (mi_str != NULL) {
+		interface_number = strtoul(&mi_str[3], &endptr, 16);
+	}
+    if (mi_str == NULL || errno != 0 || endptr - &mi_str[3] != 2) {
```

Comment:
Since in our case we always want to see 2 characters converted and there is no possibility of overflow value, if you check the end pointer, there is no need to additionally check errno.

Reply:
Removed